### DISCUSSION TITLE
Reducing outliers and updating topics when using OpenAI model as representation model

### DISCUSSION THREAD

**Discussion by serenalotreck** [MaartenGr/BERTopic#1780] [Q&A] [answered]
I've trained a model that generates the topic representations with `gpt-3.5-turbo`, using the following code:

```
sentence_model = SentenceTransformer('allenai/scibert_scivocab_cased')
umap_model = UMAP(n_neighbors=15, n_components=5, min_dist=0.0, metric='cosine', random_state=42)
vectorizer_model = CountVectorizer(stop_words="english", ngram_range=(1, 3), min_df=10)
client = openai.OpenAI(api_key=API_KEY)
representation_model = OpenAI(client, "gpt-3.5-turbo", chat=True)

topic_model = BERTopic(embedding_model=sentence_model, umap_model=umap_model, representation_model=representation_model, vectorizer_model=vectorizer_model)
topics, probs = topic_model.fit_transform(docs)
```
The output of `topic_model.get_topic_info()[:10]`:

![image](https://github.com/MaartenGr/BERTopic/assets/41377532/c04114ab-5958-41a2-9944-c73598700b9e)

However, when I try and run `reduce_outliers` and then `update_topics`, it reverts the topic representations to the c-tf-idf representations. I've tried these three things thus far:

```
# Passing the vectorizer with update topics
new_topics = topic_model.reduce_outliers(docs, topics)
topic_model.update_topics(docs, topics=new_topics, vectorizer_model=vectorizer_model)

# Not passing the vectorizer
new_topics = topic_model.reduce_outliers(docs, topics)
topic_model.update_topics(docs, topics=new_topics)

# Using embedding strategy
new_topics = topic_model.reduce_outliers(docs, topics, strategy='embeddings')
topic_model.update_topics(docs, topics=new_topics)
```

For all three options, the output looks something like this:
![image](https://github.com/MaartenGr/BERTopic/assets/41377532/cce44645-3edc-477a-a663-7a6f14402990)

I also get the following warning upon running `update_topics`:
```
2024-01-31 18:21:22,790 - BERTopic - WARNING: Using a custom list of topic assignments may lead to errors if topic reduction techniques are used afterwards. Make sure that manually assigning topics is the last step in the pipeline. Note that topic embeddings will also be created through weighted c-TF-IDF embeddings instead of centroid embeddings.
```

I assume that the issue is that the ChatGPT topics are assigned as custom topics when I first do `fit_transform`.

How can I reduce outliers while maintaining the OpenAI representations?

**Comment by MaartenGr** [comment, marked as answer]
Check the docstrings of `.update_topics`, you will notice that you are missing some vital parameters in your code, like `representation_model` and `vectorizer_model`.

**Reply by serenalotreck** [reply]
Passing `representation_model` fixed it, thank you!
